azure-pipelines.yml

```diff
@@ -6,69 +6,59 @@ variables:
   PIPENV_NOSPIN: 1
 
 jobs:
-- job: test
-  displayName: 'Lint & Test'
-
-  pool:
-    vmImage: ubuntu-16.04
-
-  variables:
-    PIPENV_CACHE_DIR: ".cache/pipenv"
-    PIP_CACHE_DIR: ".cache/pip"
-    PIP_SRC: ".cache/src"
-
-  steps:
-  - script: |
-      sudo apt-get update
-      sudo apt-get install build-essential curl docker libffi-dev libfreetype6-dev libxml2 libxml2-dev libxslt1-dev zlib1g zlib1g-dev
-    displayName: 'Install base dependencies'
-
-  - task: UsePythonVersion@0
-    displayName: 'Set Python version'
-    inputs:
-      versionSpec: '3.7.x'
-      addToPath: true
-
-  - script: sudo pip install pipenv
-    displayName: 'Install pipenv'
-
-  - script: pipenv install --dev --deploy --system
-    displayName: 'Install project using pipenv'
-
-  - script: python -m flake8
-    displayName: 'Run linter'
-
-  - script: BOT_API_KEY=foo BOT_TOKEN=bar WOLFRAM_API_KEY=baz python -m pytest --junitxml=junit.xml --cov=bot --cov-branch --cov-report=term --cov-report=xml tests
-    displayName: Run tests
-
-  - task: PublishCodeCoverageResults@1
-    displayName: 'Publish Coverage Results'
-    condition: succeededOrFailed()
-    inputs:
-      codeCoverageTool: Cobertura
-      summaryFileLocation: coverage.xml
-
-  - task: PublishTestResults@2
-    displayName: 'Publish Test Results'
-    condition: succeededOrFailed()
-    inputs:
-      testResultsFiles: junit.xml
-      testRunTitle: 'Bot Test results'
-
-- job: build
-  displayName: 'Build Containers'
-  dependsOn: 'test'
-
-  steps:
-  - task: Docker@1
-    displayName: 'Login: Docker Hub'
-
-    inputs:
-      containerregistrytype: 'Container Registry'
-      dockerRegistryEndpoint: 'DockerHub'
-      command: 'login'
-
-  - task: ShellScript@2
-    displayName: 'Build and deploy containers'
-    inputs:
-      scriptPath: scripts/deploy-azure.sh
+  - job: test
+    displayName: 'Lint & Test'
+    pool:
+      vmImage: ubuntu-16.04
+
+    variables:
+      PIP_CACHE_DIR: ".cache/pip"
+
+    steps:
+      - task: UsePythonVersion@0
+        displayName: 'Set Python version'
+        inputs:
+          versionSpec: '3.7.x'
+          addToPath: true
+
+      - script: sudo pip install pipenv
```

Comment:
Using `sudo` makes it use Python 2.7 because I suppose the `PATH` is different for the root user. This can be confirmed by looking at the logs and seeing Python 2.7 in the directories. I suppose it's fine for pipenv to be installed for 2.7 but it just seems unnecessary to even use sudo.

Reply:
Agreed. I left it in due to there being a previous commit explicitly adding it, but we don't use it elsewhere. This is an anomaly.